nengo_gui/components/spa_similarity.py

```diff
@@ -0,0 +1,102 @@
+from .pointer import Pointer
+from nengo_gui.components.component import Component
+
+import numpy as np
+import nengo
+
+import struct
+import copy
+
+class SpaSimilarity(Pointer):
+
+    config_defaults = dict(max_value=1,
+                           min_value=-1,
+                           show_pairs=False,
+                           **Component.config_defaults)
+
+    def __init__(self, obj, **kwargs):
+        super(SpaSimilarity, self).__init__(obj, **kwargs)
+        try:
+            target_key = kwargs['target']
+        except KeyError:
+            target_key = kwargs['args']
+
+        self.old_vocab_length = len(self.vocab_out.keys)
+        self.old_pairs_length = 0
+        self.labels = self.vocab_out.keys
+        self.previous_pairs = False
+
+    def gather_data(self, t, x):
+        vocab = self.vocab_out
+
+        # if there's been a change in the show_pairs
+        # Don't know how I feel about the timing of this
+        # maybe trigger on_message?
```

Comment:
Not sure about this comment. If it stays it should probably explain what the issue with the timing is.

Also the line above: Do we care about capitalization and punctuation?